In [14]:
from pathlib import Path

import numpy as np
import pandas as pd

In [15]:
MY_TABLE_STYLES = [
    {
        "selector": "th",
        "props": [
            ("background-color", "lightgrey"),
            ("font-size", "9px"),
            ("font-family", "sans-serif"),
            ("text-align", "left"),
            ("word-wrap", "break-word"),
            ("padding", "6px 10px"),
        ],
    },
    # Data cells: white background, smaller font, tight padding, NO WORD WRAP
    {
        "selector": "td",
        "props": [
            ("background-color", "white"),
            ("font-size", "9px"),
            ("font-family", "sans-serif"),
            ("white-space", "nowrap"),  # <-- Prevents ugly text wrapping
            ("padding", "4px 10px"),  # <-- Tighter padding makes rows shorter
        ],
    },
    # Alternating row colors (Zebra Striping / nth-child)
    {"selector": "tr:nth-child(even) td", "props": [("background-color", "#f9f9f9")]},
    # Hover effect
    {"selector": "tr:hover td", "props": [("background-color", "lightblue")]},
]


def show_table(df, rows=5, title=None):
    if title:
        print(f"=== {title.upper()} ===")

    preview = df.head(rows)

    num_cols = preview.select_dtypes(include=np.number).columns

    return (
        preview.style.set_table_styles(MY_TABLE_STYLES)
        .hide(axis="index")
        .set_properties(subset=num_cols, **{"text-align": "right"})
        .highlight_null(color="rgba(255, 0, 0, 0.3)")
        .format(precision=2)
    )


def show_schema(df):
    schema = df.dtypes.rename("dtype").to_frame().reset_index(names="column")

    return schema.style.set_table_styles(MY_TABLE_STYLES).hide(axis="index")

In [17]:
MARCH_RAW_DIR = Path("../data/raw/pcdd/2026-03")
JUNE_RAW_DIR = Path("../data/raw/pcdd/2026-06")

june_paths = {
    "rate": JUNE_RAW_DIR / "pcdem-nhs-rate-jun-2026.csv",
    "sub_icb": JUNE_RAW_DIR / "pcdem-sub-icb-jun-2026.csv",
    "practice": JUNE_RAW_DIR / "pcdem-practice-jun-2026.csv",
    "mapping": JUNE_RAW_DIR / "mapping-file-july-dementia-2026.csv",
    "dictionary": JUNE_RAW_DIR / "PCDD-2627-data-dictionary.xlsx",
}

rate = pd.read_csv(june_paths["rate"])
sub_icb = pd.read_csv(june_paths["sub_icb"])
practice = pd.read_csv(june_paths["practice"])
mapping = pd.read_csv(june_paths["mapping"])
dictionary = pd.read_excel(june_paths["dictionary"])

datasets = {"rate": rate, "sub_icb": sub_icb, "practice": practice, "mapping": mapping}

for name, dataset in datasets.items():
    print(f"Displaying dataset: {name}")
    # display(show_table(dataset))
    # display(show_schema(dataset))
    display(dataset)
    print(f"shape of {name}: {dataset.shape}")
    print()

Displaying dataset: rate


,INDICATOR,ORG_TYPE,ORG_CODE,ONS_CODE,NAME,ACH_DATE,MEASURE,VALUE,DQ
0,DEMENTIA: 65+ ESTIMATED DIAGNOSIS RATE,COUNTRY_RESPONSIBILITY,ENG,ENG,ENGLAND,30-Jun-26,DEMENTIA_ESTIMATE_65_PLUS,756824.7,NaN
1,DEMENTIA: 65+ ESTIMATED DIAGNOSIS RATE,COUNTRY_RESPONSIBILITY,ENG,ENG,ENGLAND,30-Jun-26,DEMENTIA_REGISTER_65_PLUS,501216.0,NaN
2,DEMENTIA: 65+ ESTIMATED DIAGNOSIS RATE,COUNTRY_RESPONSIBILITY,ENG,ENG,ENGLAND,30-Jun-26,DIAG_RATE_65_PLUS,66.2,NaN
3,DEMENTIA: 65+ ESTIMATED DIAGNOSIS RATE,COUNTRY_RESPONSIBILITY,ENG,ENG,ENGLAND,30-Jun-26,DIAG_RATE_65_PLUS_LL,59.6,NaN
4,DEMENTIA: 65+ ESTIMATED DIAGNOSIS RATE,COUNTRY_RESPONSIBILITY,ENG,ENG,ENGLAND,30-Jun-26,DIAG_RATE_65_PLUS_UL,71.6,NaN
...,...,...,...,...,...,...,...,...,...
745,DEMENTIA: 65+ ESTIMATED DIAGNOSIS RATE,SUB_ICB_LOC,X2C4Y,E38000254,NHS WEST YORKSHIRE ICB - X2C4Y,30-Jun-26,DEMENTIA_ESTIMATE_65_PLUS,5172.2,NaN
746,DEMENTIA: 65+ ESTIMATED DIAGNOSIS RATE,SUB_ICB_LOC,X2C4Y,E38000254,NHS WEST YORKSHIRE ICB - X2C4Y,30-Jun-26,DEMENTIA_REGISTER_65_PLUS,3431.0,NaN
747,DEMENTIA: 65+ ESTIMATED DIAGNOSIS RATE,SUB_ICB_LOC,X2C4Y,E38000254,NHS WEST YORKSHIRE ICB - X2C4Y,30-Jun-26,DIAG_RATE_65_PLUS,66.3,NaN
748,DEMENTIA: 65+ ESTIMATED DIAGNOSIS RATE,SUB_ICB_LOC,X2C4Y,E38000254,NHS WEST YORKSHIRE ICB - X2C4Y,30-Jun-26,DIAG_RATE_65_PLUS_LL,59.3,NaN


shape of rate: (750, 9)

Displaying dataset: sub_icb


,ACH_DATE,ORG_TYPE,ODS_CODE,ONS_CODE,NAME,MEASURE,BREAKDOWN,AGE,GENDER,ETHNICITY,DEMENTIA_TYPE,RESIDENTIAL_TYPE,VALUE
0,30-Jun-26,SUB_ICB,00L,E38000130,NHS NORTH EAST AND NORTH CUMBRIA ICB - 00L,DELIRIUM_12M,NaN,ALL,ALL,ALL,ALL,ALL,262
1,30-Jun-26,SUB_ICB,00L,E38000130,NHS NORTH EAST AND NORTH CUMBRIA ICB - 00L,DEMENTIA_REGISTER,AGE_GENDER,0_39,Female,ALL,ALL,ALL,2
2,30-Jun-26,SUB_ICB,00L,E38000130,NHS NORTH EAST AND NORTH CUMBRIA ICB - 00L,DEMENTIA_REGISTER,AGE_GENDER,40_44,Female,ALL,ALL,ALL,0
3,30-Jun-26,SUB_ICB,00L,E38000130,NHS NORTH EAST AND NORTH CUMBRIA ICB - 00L,DEMENTIA_REGISTER,AGE_GENDER,45_49,Female,ALL,ALL,ALL,2
4,30-Jun-26,SUB_ICB,00L,E38000130,NHS NORTH EAST AND NORTH CUMBRIA ICB - 00L,DEMENTIA_REGISTER,AGE_GENDER,50_54,Female,ALL,ALL,ALL,3
...,...,...,...,...,...,...,...,...,...,...,...,...,...
11655,30-Jun-26,SUB_ICB,X2C4Y,E38000254,NHS WEST YORKSHIRE ICB - X2C4Y,PRESCRIBING,PSY_DIAG,ALL,ALL,ALL,ALL,ALL,172
11656,30-Jun-26,SUB_ICB,X2C4Y,E38000254,NHS WEST YORKSHIRE ICB - X2C4Y,PRESCRIBING,PSY_DIAG_WITH_ANTIPSYMEDS,ALL,ALL,ALL,ALL,ALL,104
11657,30-Jun-26,SUB_ICB,X2C4Y,E38000254,NHS WEST YORKSHIRE ICB - X2C4Y,REFERRALS,NUM_PAT_REF_MEM_CLIN_12M,ALL,ALL,ALL,ALL,ALL,718
11658,30-Jun-26,SUB_ICB,X2C4Y,E38000254,NHS WEST YORKSHIRE ICB - X2C4Y,REFERRALS,NUM_REF_MEM_CLIN_12M,ALL,ALL,ALL,ALL,ALL,849


shape of sub_icb: (11660, 13)

Displaying dataset: practice


,ACH_DATE,ORG_TYPE,ODS_CODE,NAME,MEASURE,BREAKDOWN,VALUE
0,30-Jun-26,PRACTICE,A81001,THE DENSHAM SURGERY,DEMENTIA_REGISTER,DEMENTIA_REGISTER_0_64,0
1,30-Jun-26,PRACTICE,A81001,THE DENSHAM SURGERY,DEMENTIA_REGISTER,DEMENTIA_REGISTER_65_PLUS,28
2,30-Jun-26,PRACTICE,A81001,THE DENSHAM SURGERY,PAT_LIST,PAT_LIST_0_64,2850
3,30-Jun-26,PRACTICE,A81001,THE DENSHAM SURGERY,PAT_LIST,PAT_LIST_65_PLUS,883
4,30-Jun-26,PRACTICE,A81001,THE DENSHAM SURGERY,REVIEWS,DIAG_DECLINED_CARE_PLAN,0
...,...,...,...,...,...,...,...
42611,30-Jun-26,PRACTICE,Y08411,ONE CARE LEWISHAM,PAT_LIST,PAT_LIST_0_64,38
42612,30-Jun-26,PRACTICE,Y08411,ONE CARE LEWISHAM,PAT_LIST,PAT_LIST_65_PLUS,288
42613,30-Jun-26,PRACTICE,Y08411,ONE CARE LEWISHAM,REVIEWS,DIAG_DECLINED_CARE_PLAN,0
42614,30-Jun-26,PRACTICE,Y08411,ONE CARE LEWISHAM,REVIEWS,DIAG_RECEIVED_CARE_PLAN,17


shape of practice: (42616, 7)

Displaying dataset: mapping


,EXTRACT_DATE,PRACTICE_CODE,PRACTICE_NAME,PRACTICE_POSTCODE,ONS_SUB_ICB_LOCATION_CODE,SUB_ICB_LOCATION_CODE,SUB_ICB_LOCATION_NAME,ONS_ICB_CODE,ICB_CODE,ICB_NAME,ONS_COMM_REGION_CODE,COMM_REGION_CODE,COMM_REGION_NAME
0,01-Jul-26,A81001,THE DENSHAM SURGERY,TS18 1HU,E38000247,16C,NHS NORTH EAST AND NORTH CUMBRIA ICB - 16C,E54000050,QHM,NHS NORTH EAST AND NORTH CUMBRIA INTEGRATED CA...,E40000012,Y63,NORTH EAST AND YORKSHIRE COMMISSIONING REGION
1,01-Jul-26,A81002,QUEENS PARK MEDICAL CENTRE,TS18 2AW,E38000247,16C,NHS NORTH EAST AND NORTH CUMBRIA ICB - 16C,E54000050,QHM,NHS NORTH EAST AND NORTH CUMBRIA INTEGRATED CA...,E40000012,Y63,NORTH EAST AND YORKSHIRE COMMISSIONING REGION
2,01-Jul-26,A81004,ACKLAM MEDICAL CENTRE,TS5 8SB,E38000247,16C,NHS NORTH EAST AND NORTH CUMBRIA ICB - 16C,E54000050,QHM,NHS NORTH EAST AND NORTH CUMBRIA INTEGRATED CA...,E40000012,Y63,NORTH EAST AND YORKSHIRE COMMISSIONING REGION
3,01-Jul-26,A81005,SPRINGWOOD SURGERY,TS14 7DJ,E38000247,16C,NHS NORTH EAST AND NORTH CUMBRIA ICB - 16C,E54000050,QHM,NHS NORTH EAST AND NORTH CUMBRIA INTEGRATED CA...,E40000012,Y63,NORTH EAST AND YORKSHIRE COMMISSIONING REGION
4,01-Jul-26,A81006,TENNANT STREET MEDICAL PRACTICE,TS18 2AT,E38000247,16C,NHS NORTH EAST AND NORTH CUMBRIA ICB - 16C,E54000050,QHM,NHS NORTH EAST AND NORTH CUMBRIA INTEGRATED CA...,E40000012,Y63,NORTH EAST AND YORKSHIRE COMMISSIONING REGION
...,...,...,...,...,...,...,...,...,...,...,...,...,...
6177,01-Jul-26,Y08411,ONE CARE LEWISHAM,SE23 1RJ,E38000244,72Q,NHS SOUTH EAST LONDON ICB - 72Q,E54000030,QKK,NHS SOUTH EAST LONDON INTEGRATED CARE BOARD,E40000003,Y56,LONDON COMMISSIONING REGION
6178,01-Jul-26,Y08516,YORK GP EXTENDED ACCESS (NIMBUSCARE),YO24 4LJ,E38000188,03Q,NHS HUMBER AND NORTH YORKSHIRE ICB - 03Q,E54000051,QOQ,NHS HUMBER AND NORTH YORKSHIRE INTEGRATED CARE...,E40000012,Y63,NORTH EAST AND YORKSHIRE COMMISSIONING REGION
6179,01-Jul-26,Y08645,WAVERLEY MEDICAL CENTRE,S60 8GD,E38000141,03L,NHS SOUTH YORKSHIRE ICB - 03L,E54000061,QF7,NHS SOUTH YORKSHIRE INTEGRATED CARE BOARD,E40000012,Y63,NORTH EAST AND YORKSHIRE COMMISSIONING REGION
6180,01-Jul-26,Y08852,STAYCITY YORK (NIMBUSCARE),YO24 4LJ,E38000188,03Q,NHS HUMBER AND NORTH YORKSHIRE ICB - 03Q,E54000051,QOQ,NHS HUMBER AND NORTH YORKSHIRE INTEGRATED CARE...,E40000012,Y63,NORTH EAST AND YORKSHIRE COMMISSIONING REGION


shape of mapping: (6182, 13)



In [ ]:
xls = pd.ExcelFile(paths["dictionary"])
display(xls.sheet_names)

['Title Sheet',
 'Notes and caveats',
 'pcdem-nhs-rate',
 'pcdem-la-rate',
 'pcdem-practice',
 'pcdem-sub-icb',
 'Age_Gender categories',
 'Ethnicity categories',
 'Dementia Type categories',
 'Residential Type categories',
 'mapping-file-mmm-dementia-yyyy']

## 2. Dataset structure and grain

Before analysing values, establish what one row represents in each dataset and whic columns uniquely indentify a record.

In [53]:
for name, df in datasets.items():
    print(f"\n{name.upper()}")
    print((f"Rows: {len(df):,}"))
    print(f"Columns: {len(df.columns)}")
    print("Columns:")
    print(df.columns.tolist())
    


RATE
Rows: 750
Columns: 9
Columns:
['INDICATOR', 'ORG_TYPE', 'ORG_CODE', 'ONS_CODE', 'NAME', 'ACH_DATE', 'MEASURE', 'VALUE', 'DQ']

SUB_ICB
Rows: 11,660
Columns: 13
Columns:
['ACH_DATE', 'ORG_TYPE', 'ODS_CODE', 'ONS_CODE', 'NAME', 'MEASURE', 'BREAKDOWN', 'AGE', 'GENDER', 'ETHNICITY', 'DEMENTIA_TYPE', 'RESIDENTIAL_TYPE', 'VALUE']

PRACTICE
Rows: 42,616
Columns: 7
Columns:
['ACH_DATE', 'ORG_TYPE', 'ODS_CODE', 'NAME', 'MEASURE', 'BREAKDOWN', 'VALUE']

MAPPING
Rows: 6,182
Columns: 13
Columns:
['EXTRACT_DATE', 'PRACTICE_CODE', 'PRACTICE_NAME', 'PRACTICE_POSTCODE', 'ONS_SUB_ICB_LOCATION_CODE', 'SUB_ICB_LOCATION_CODE', 'SUB_ICB_LOCATION_NAME', 'ONS_ICB_CODE', 'ICB_CODE', 'ICB_NAME', 'ONS_COMM_REGION_CODE', 'COMM_REGION_CODE', 'COMM_REGION_NAME']


In [20]:
rate["DQ"].isna().mean()
rate["DQ"].value_counts(dropna=False)

DQ
NaN    750
Name: count, dtype: int64

## Rate Dataset

**Publisher definition:** Diagnosis rate indicator by sub ICB location - ICB - region - England

**Shape:** 750 rows × 9 columns

### What does one row represent?

The rate dataset reports dementia measures for NHS organisations/georgaphies at the reporting period end date.

Key fields:
- `NAME` - organisation name
    - an organisation is a boundary definition such as 'ENGLAND' or 'NHS NORTH WEST YORKSHIRE ICB - X2C4Y
- `ACH_DATE` - achieved date
    - the achieved date represnts the reporting period end date
    - this terminology is inherited from the Quality and Outcomes Framework (QOF) system
- `MEASURE` - identifies what the `VALUE` represents
- `VALUE` - alue associated with the measure
- `ORG_TYPE` - organisation breakdown (geographic level)
- `ORG_CODE` - ODS (Organisation Data Service) organisation identifier
- `ONS_CODE` - 9-digit geography code of the organisation
- `DQ` - "Identifies organisations with a smaller denominator population than the CFAS II reference population with a ‘1’. Values for these organisations should be interpreted with caution."

Key measures:
|Measure|Meaning|
|---|---|
| DEMENTIA_REGISTER_65_PLUS | Number of patients aged 65 and over with a recorded dementia diagnosis |
| DEMENTIA_ESTIMATE_65_PLUS | Expected number of patients aged 65 and over with dementia |
| DIAG_RATE_65_PLUS | Estimated dementia diagnosis rate (recorded/expected) in patients aged 65 and over |
| DIAG_RATE_65_PLUS_LL | Lower 95% confidence limit of estimated dementia diagnosis rate in patients aged 65 and over |
| DIAG_RATE_65_PLUS_UL | Upper 95% confidence limit of estimated dementia diagnosis rate in patients aged 65 and over |



### ORG_TYPE

In [61]:
rate["ORG_TYPE"].value_counts().to_frame()

,count
ORG_TYPE,
SUB_ICB_LOC,530
ICB,180
NHS_REGION,35
COUNTRY_RESPONSIBILITY,5


In [60]:
rate.groupby("ORG_TYPE")["ORG_CODE"].nunique().to_frame()

,ORG_CODE
ORG_TYPE,
COUNTRY_RESPONSIBILITY,1
ICB,36
NHS_REGION,7
SUB_ICB_LOC,106


In [62]:
# Calculate unique entities by dividing row counts by the 5 measure types
unique_org_counts = (rate["ORG_TYPE"].value_counts() / 5).astype(int)

# Display as a clean DataFrame
print("Unique entity count for geographic levels (organisations)")
unique_org_counts.to_frame(name="UNIQUE_ORG_COUNT")


Unique entity count for geographic levels (organisations)


,UNIQUE_ORG_COUNT
ORG_TYPE,
SUB_ICB_LOC,106
ICB,36
NHS_REGION,7
COUNTRY_RESPONSIBILITY,1


### MEASURE

In [47]:
display(rate["MEASURE"].value_counts().to_frame())

,count
MEASURE,
DEMENTIA_ESTIMATE_65_PLUS,150
DEMENTIA_REGISTER_65_PLUS,150
DIAG_RATE_65_PLUS,150
DIAG_RATE_65_PLUS_LL,150
DIAG_RATE_65_PLUS_UL,150


Don't need to check the other fields as they will just show 150 unique items for the 5 measure types, as well as dates, as well as NaN for DQ

### Rate Dataset Structure Breakdown

The `rate` dataframe contains **750 rows** in total. Because each geographical entity is repeated exactly **5 times** (once for each of the 5 diagnostic measure types), dividing the total row counts by 5 reveals the true number of unique reporting organizations in the dataset:

| Organization Type (`ORG_TYPE`) | Total Rows in File | Divided by 5 | Unique Reporting Entities |
| :--- | :--- | :--- | :--- |
| **SUB_ICB_LOC** | 530 | ÷ 5 | **106** Sub-ICB Locations |
| **ICB** | 180 | ÷ 5 | **36** Integrated Care Boards |
| **NHS_REGION** | 35 | ÷ 5 | **7** NHS Regions |
| **COUNTRY_RESPONSIBILITY** | 5 | ÷ 5 | **1** Country (`ENGLAND`) |
| **Total** | **750** | | **150 Unique Entities** |


## Sub-ICB Dataset

**Publisher definition:** Primary care dementia data: Dementia data by sub ICB location

### What does one row represent?

The Sub-ICB dataset reports a measure/value for a Sub-ICB at the reporting period end date, with some measures further broken down by dimensions such as age, gender, ethnicity, dementia type or residential type.

Key fields:

- `ACH_DATE` - achieved date
    - represents the reporting period end date
    - this terminology is inherited from the Quality and Outcomes Framework (QOF) system

- `MEASURE` - identifies what the `VALUE` represents

- `BREAKDOWN` - identifies whether/how the measure has been broken down into a more specific group
    - e.g. `AGE_GENDER` indicates that the measure is split by age and gender

- `AGE` - age group for the observation
    - `ALL` indicates the measure is not restricted to a particular age group

- `GENDER` - gender group for the observation
    - `ALL` indicates the measure is not restricted to a particular gender

- `ETHNICITY` - ethnicity group for the observation
    - `ALL` indicates the measure is not restricted to a particular ethnicity group

- `DEMENTIA_TYPE` - dementia type for the observation
    - `ALL` indicates the measure is not restricted to a particular dementia type

- `RESIDENTIAL_TYPE` - residential setting/type for the observation
    - `ALL` indicates the measure is not restricted to a particular residential type

- `VALUE` - value associated with the measure and its applicable breakdown

### ORG / Sub-ICB entities

### MARKDOWN

In [66]:
sub_icb["MEASURE"].value_counts()

MEASURE
DEMENTIA_REGISTER    4876
PAT_LIST             2544
MCI                  2332
PRESCRIBING           742
FRAILTY               530
REFERRALS             212
DELIRIUM_12M          106
INCIDENCE             106
PALLIATIVE_CARE       106
YOUNG_ONSET           106
Name: count, dtype: int64

### BREAKDOWN

In [67]:
sub_icb["BREAKDOWN"].value_counts(dropna=False)

BREAKDOWN
AGE_GENDER                      7420
DEMENTIA_TYPE                    848
ETHNICITY                        848
RESIDENCE_TYPE                   636
NaN                              424
DEM_FALL_12M                     106
DEM_FRAC_NOF_12M                 106
DEM_MILD_FRAILTY                 106
DEM_MOD_FRAILTY                  106
DEM_SEV_FRAILTY                  106
DON_RIV_GAL_12M                  106
MEM_12M                          106
MEM_PLUS_DON_RIV_GAL_12M         106
NO_PSY_DIAG                      106
NO_PSY_DIAG_WITH_ANTIPSYMEDS     106
PSY_DIAG                         106
PSY_DIAG_WITH_ANTIPSYMEDS        106
NUM_PAT_REF_MEM_CLIN_12M         106
NUM_REF_MEM_CLIN_12M             106
Name: count, dtype: int64

In [68]:
sub_icb[
    ["MEASURE", "BREAKDOWN"]
].drop_duplicates().sort_values(
    ["MEASURE", "BREAKDOWN"],
    na_position="first",
)

,MEASURE,BREAKDOWN
0,DELIRIUM_12M,NaN
1,DEMENTIA_REGISTER,AGE_GENDER
25,DEMENTIA_REGISTER,DEMENTIA_TYPE
33,DEMENTIA_REGISTER,ETHNICITY
41,DEMENTIA_REGISTER,RESIDENCE_TYPE
47,FRAILTY,DEM_FALL_12M
48,FRAILTY,DEM_FRAC_NOF_12M
49,FRAILTY,DEM_MILD_FRAILTY
50,FRAILTY,DEM_MOD_FRAILTY
51,FRAILTY,DEM_SEV_FRAILTY


### AGE / GENDER

In [70]:
sub_icb["AGE"].value_counts()

AGE
ALL        4240
40_44       636
45_49       636
50_54       636
55_59       636
60_64       636
65_69       636
70_74       636
75_79       636
80_84       636
85_89       636
90_PLUS     636
0_39        424
Name: count, dtype: int64

In [71]:
sub_icb["GENDER"].value_counts()

GENDER
ALL       4240
Female    2438
Male      2438
FEMALE    1272
MALE      1272
Name: count, dtype: int64

### ETHNICITY

In [72]:
sub_icb["ETHNICITY"].value_counts()

ETHNICITY
ALL                                               10812
ASIAN_OR_ASIAN_BRITISH                              106
BLACK_OR_AFRICAN_OR_CARIBBEAN_OR_BLACK_BRITISH      106
INCONCLUSIVE_ETHNIC_GROUP                           106
MIXED_OR_MULTIPLE_ETHNIC_GROUPS                     106
NOT_DEFINED                                         106
NOT_STATED                                          106
OTHER_ETHNIC_GROUP                                  106
WHITE                                               106
Name: count, dtype: int64

### DEMENTIA TYPE

In [73]:
sub_icb["DEMENTIA_TYPE"].value_counts()

DEMENTIA_TYPE
ALL                                 10812
ALZHEIMERS_DISEASE                    106
FRONTOTEMPORAL                        106
INCONCLUSIVE_DEMENTIA_TYPE            106
LEWY_BODY                             106
MIXED_DEMENTIA_TYPES                  106
OTHER_SPECIFIED_DEMENTIA_TYPES        106
OTHER_UNSPECIFIED_DEMENTIA_TYPES      106
VASCULAR_DEMENTIA                     106
Name: count, dtype: int64

### RESIDENTIAL TYPE

In [74]:
sub_icb["RESIDENTIAL_TYPE"].value_counts()

RESIDENTIAL_TYPE
ALL                              11024
INCONCLUSIVE_RESIDENTIAL_TYPE      106
NO_PERMANENT_ADDRESS               106
NURSING_HOME                       106
OTHER_RESIDENTIAL_TYPE             106
PRIVATE_RESIDENCE                  106
RESIDENTIAL_CARE_HOME              106
Name: count, dtype: int64

### Sub-ICB Dataset Structure Breakdown

The dataset contains 11,660 observations for Sub-ICB locations.

Unlike the rate dataset, each Sub-ICB does not have a fixed number of rows.
The number of records depends on the measures reported and the dimensional
breakdowns attached to those measures.

Some measures contain a single overall value, while others are expanded
across categories such as age/gender, ethnicity, dementia type or residential type.